specs/~lambdas.json

```diff
@@ -15,7 +15,8 @@
           "php": "return \"world\";",
           "python": "lambda: \"world\"",
           "clojure": "(fn [] \"world\")",
-          "lisp": "(lambda () \"world\")"
+          "lisp": "(lambda () \"world\")",
+          "pwsh": "\"world\""
```

Comment:
Is this really PowerShell's equivalent of a lambda? Shouldn't there be something like a `function` around the expressions? Or perhaps `echo` or `return` in front?

Reply:
Hi Julian,

no, it's not necessary. Unlike other languages PowerShell returns the output of each expression, when the output is not captures in any way e.g. with a variable assignment. The return statement is optional in this case.

You can find more about this behavior here https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_return?view=powershell-7.1.

PowerShell does not have lambda functions in the way you may know them, but the nearest equivalent are ScriptBlocks which works similar. (https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_script_blocks?view=powershell-7.1)

~~~PowerShell
PS> $foo = { "Hello" }
PS> Invoke-Command -ScriptBlock $foo
Hello

PS> $foo = { $args[0] + 1 }
PS> Invoke-Command -ScriptBlock $foo -ArgumentList 1
2
~~~

Reply:
So in this specific case, `lambda` will behave as a property that dynamically returns `"world"` (when invoked as a function/method) rather than just acting as a property that has this value statically?